strawberryfields/backends/fockbackend/ops.py

```diff
@@ -328,6 +329,20 @@ def beamsplitter(theta, phi, trunc):
     return BS_tw.transpose((0, 2, 1, 3))
 
 
+@functools.lru_cache()
```

Comment:
Does it make sense to use `lru_cache` with `float` inputs? Will the cache ever get successful hits?

Reply:
Unless the parameters are hashed with some sort of rounding this does not make sense indeed. 
However, I see that on the same file some other functions accepting `float` params have the annotation.

Reply:
@ziofil : keep in mind the number of times that you can potentially call a 50/50 beamsplitter, `BSgate(np.pi/4, 0)`

Reply:
ok I checked and floats are hit correctly with lru_cache